File: Exercitar java/Exercicio08/Exercicio08.java

import java.lang.Math;


class Main{
    
    /*
        Função dada:
            f(x) = x^3 - x^2 - 13x + 8
    */ 
    static double f(double x){
        return ( (x*x*x) - (x*x) -(13*x) +8 );
    }

    /*
        Derivada da função dada:
            f'(x) = 3x^2 - 2x  - 13
    */ 
    static double f_linha(double x){
        return ( (3*x*x) - 2*x -13 );
    }
    
    public static void main(String[] args) throws Exception {

        //Leitura e declaração inicial
        System.out.printf("Insira o chute inicial:\n");
        double raiz_chute = EntradaTeclado.leDouble();
        double raiz = 0;

        //Primeira iteração
        raiz = raiz_chute - (f(raiz_chute)/f_linha(raiz_chute));
        
        //Loop de calculo
        while(Math.abs(raiz_chute - raiz) > 0.00000001){
            System.out.printf("===== Entrou no loop =====\n");
            System.out.printf("Raiz chute %f |-| Raiz %f\n");
            
            raiz_chute = raiz;
            raiz = raiz_chute - (f(raiz_chute)/f_linha(raiz_chute));

            System.out.printf("======== Fim do loop =======\n");
            System.out.printf("Raiz chute %f |-| Raiz %f\n");

        }



    }
}

File: Exercitar python/Exercicio08.py
def modulo(x):
    if(x < 0): return (-(x))
    else: return x

#Função dada
#f(x) = x^3 - x^2 - 13x + 8
def f(x):
    return ( (x*x*x) - (x*x) -(13*x) + 8 )

#Derivada da função dada:
#f'(x) = 3x^2 - 2x  - 13
def f_linha(x):
    return ( (3*x*x) - (2*x) - 13 )

#Input inicial
print("Insira o valor do chute inicial")
chute_inicial = float(input())
raiz_anterior = chute_inicial

#Calculo inicial
raiz_atual = raiz_anterior - ( f(raiz_anterior)/f_linha(raiz_anterior))
raiz_anterior = raiz_atual

#Loop de resolução
while(modulo(raiz_atual - raiz_anterior) > 0.00000001):
    raiz_atual = raiz_anterior - ( f(raiz_anterior)/f_linha(raiz_anterior))
    raiz_anterior = raiz_atual

#Saída do programa
print("O valor da raíz é",raiz_atual)
